clippy_lints/src/methods/mod.rs

```diff
@@ -435,7 +435,18 @@ declare_clippy_lint! {
     /// ```no_run
     /// # struct X;
     /// impl X {
-    ///     fn as_str(self) -> &'static str {
+    ///     fn str(&self) -> &'static str {
```

Comment:
This no longer actually triggers the lint, does it? I think the old example was good for showing what not to write, with `fn as_str(self)`. Maybe we can keep the old example here and your new example for what to write (`fn as_str(&self)`) instead?

Reply:
Make sense